lib/checkassert.cpp

```diff
@@ -129,6 +129,9 @@ void CheckAssert::checkVariableAssignment(const Token* assignTok, const Scope *a
     if (!assignTok->isAssignmentOp() && assignTok->tokType() != Token::eIncDecOp)
         return;
 
+    if (!assignTok->astOperand1())
```

Comment:
I think `=` must always have both operands, so we should add this to the AST validation.

Reply:
Unfortuantely, at least these cases would require special handling though:
~~~
auto f() { [=]() { return data; }(); }
~~~
~~~
class A {
    virtual void pure()=0;
};
~~~

Reply:
So we leave it as is?

Reply:
New idea: reject `=` followed by `;` and possibly `)|}` as garbage?

Reply:
Do not ask me. There's a reason I usually do not get involved into the check related stuff...

Reply:
IMHO we should fail early and hard on invalid code. Random nullptr checks here and there probably just enable the next crash down the line.

Reply:
I split this into #6162.